Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,334 @@
+            get => Parse(CurrentPerson.ModDataProcessing) + (CurrentPerson.Overclocked == "DataProcessing" ? 1 : 0);
+            set
+            {
+                if (CurrentPerson.ModDataProcessing != value.ToString())
+                {
+                    CurrentPerson.ModDataProcessing = value.ToString();
+                    OnPropertyChanged();
+                    OnPropertyChanged("TotalDataProcessing");
+                }
+            }
+        }
+
+        public int TotalDataProcessing => Parse(CurrentPerson.DataProcessing) + Parse(CurrentPerson.ModDataProcessing) + (CurrentPerson.Overclocked == "DataProcessing" ? 1 : 0);
+        
+
+        public int Firewall
+        {
+            get => Parse(CurrentPerson.Firewall);
+            set
+            {
+                if (CurrentPerson.Firewall != value.ToString())
+                {
+                    CurrentPerson.Firewall = value.ToString();
+                    OnPropertyChanged();
+                    OnPropertyChanged("TotalFirewall");
+                }
+            }
+        }
+
+        public int FirewallMod
+        {
+            get => Parse(CurrentPerson.ModFirewall) + (CurrentPerson.Overclocked == "Firewall" ? 1 : 0);
+            set
+            {
+                if (CurrentPerson.ModFirewall != value.ToString())
+                {
+                    CurrentPerson.ModFirewall = value.ToString();
+                    OnPropertyChanged();
+                    OnPropertyChanged("TotalFirewall");
+                }
+            }
+        }
+
+        public int TotalFirewall => Parse(CurrentPerson.Firewall) + Parse(CurrentPerson.ModFirewall) + (CurrentPerson.Overclocked == "Firewall" ? 1 : 0);
+        
+
+        public int ActionDicePool => GetTotalAttribute(CurrentAction.ActionAttribute) +
+                GetTotalSkill(CurrentAction.ActionSkill) +
+                CurrentAction.ActionModifier +
+                ActionModifier;
+        
+
+        public int DefenceDicePool => GetTotalSkill(CurrentAction.DefenceSkill) +
+                GetTotalAttribute(CurrentAction.DefenceAttribute) +
+                CurrentAction.DefenceModifier +
+                ActionModifier;
+        
+        public int ActionModifier
+        {
+            get => actionModifier;
+            set
+            {
+                actionModifier = value;
+                OnPropertyChanged();
+                OnPropertyChanged("ActionDicePool");
+                OnPropertyChanged("DefenceDicePool");
+            }
+        }
+
+        public int GetTotalMatrixAttribute(string attribute)
+        {
+            switch (attribute)
+            {
+                case "Attack":
+                    return this.TotalAttack;
+                case "Sleaze":
+                    return this.TotalSleaze;
+                case "Data Processing":
+                    return this.TotalDataProcessing;
+                case "Firewall":
+                    return this.TotalFirewall;
```

Comment:
You don't really need `this` here. The reason you'll find `this` in the main Chummer project in places that work off of Matrix Attributes is because of weird interactions regarding interfaces of `IHasMatrixAttribute` with a given class.

Reply:
Remove it in https://github.com/chummer5a/chummer5a/pull/4563/commits/0f130c19bc6cec27e978019f38ca192c428d1f3a